google/cloud/internal/curl_handle_factory.cc

```diff
@@ -116,22 +117,38 @@ void PooledCurlHandleFactory::CleanupHandle(CurlPtr h, HandleDisposition d) {
     std::unique_lock<std::mutex> lk(last_client_ip_address_mu_);
     last_client_ip_address_ = ip;
   }
-  if (d == HandleDisposition::kDiscard) return;
+  std::unique_lock<std::mutex> lk(handles_mu_);
+  if (d == HandleDisposition::kDiscard) {
+    --active_handles_;
+    return;
+  }
   // Use a temporary data structure to release any excess handles *after* the
   // lock is released.
   std::vector<CurlPtr> released;
```

Comment:
It looks like you'll now want to move this declaration up too.

Reply:
Doh! Fixed.